### PR TITLE
Add queue management functionality

### PR DESCRIPTION
**Pull request by bclews**
I have updated the CLI to support IVCAP Queue services (i.e. /1/queues) addressing #40  

The CLI help for `ivcap queue` now looks like:
```
Queues are used to store messages in a sequential order. You can create, read, update, and delete queues using this command. You can also add and remove messages from queues.

Usage:
  ivcap queue [command]

Aliases:
  queue, q, queues

Available Commands:
  create      Create a new queue
  delete      Delete a queue
  dequeue     Dequeue messages from a queue
  enqueue     Enqueue a message to a queue
  get         Fetch details about a single queue
  list        List existing queues

Flags:
  -h, --help   help for queue

Global Flags:
      --access-token string   Access token to use for authentication with API server [IVCAP_ACCESS_TOKEN]
      --context string        Context (deployment) to use
      --debug                 Set logging level to DEBUG
      --no-history            Do not store history
  -o, --output string         Set format for displaying output [json, yaml]
      --silent                Do not show any progress information
      --timeout int           Max. number of seconds to wait for completion (default 30)

Use "ivcap queue [command] --help" for more information about a command.
```

**List Queue**
```
List existing queues

Usage:
  ivcap queue list [flags]

Flags:
      --at-time string    query state at this time in the past
      --filter string     filter list (e.g. "name~=Fred")
  -h, --help              help for list
  -l, --limit int         max number of records to be returned (default 10)
      --order-by string   feature to order list by (e.g. "created-at,status")
      --order-desc        if set, order in descending order
  -p, --page string       page cursor
```

**Create Queue**
```
Create a new queue with the specified name. Optionally, you can also provide a description and a policy for the queue.

An example of creating a queue with a description:

  ivcap queue create --description "This is a test queue" test_queue

Usage:
  ivcap queue create [flags] name

Flags:
  -d, --description string   Description of the queue
  -h, --help                 help for create
  -p, --policy string        Policy controlling access
.
.
```

**Delete Queue**
```
Delete a queue with the specified ID. This action is irreversible. All messages in the queue will be lost.
An example of deleting a queue:

  ivcap queue delete test_queue

Usage:
  ivcap queue delete [flags] queue_id

Aliases:
  delete, remove
.
.
.
```

**Get Queue**
```
Fetch details about a single queue. You must provide the ID of the queue you want to read.

An example of reading a queue:

  ivcap queue get urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167

Usage:
  ivcap queue get [flags] queue_id

Aliases:
  get, read
.
.
.
```

**Enqueue**
```
Enqueue a message to the specified queue. The message content must be in JSON format. You must also provide the schema of the message content.

An example of enqueuing a message to a queue:

  ivcap queue enqueue urn:ivcap:schema:queue:message.1 urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167 "{\"temperature\": \"21\", \"location\": \"Buoy101\", \"timestamp\": \"2024-05-20T14:30:00Z\"}"

Usage:
  ivcap queue enqueue [flags] schema queue_id message
.
.
.
```

**Dequeue**
```
Dequeue messages from the specified queue. By default, only one message is dequeued. You can specify the number of messages to dequeue using the --limit flag. 

An example of dequeuing a message from a queue:

  ivcap queue dequeue --limit 5 urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167

Usage:
  ivcap queue dequeue [flags] queue_id

Flags:
  -h, --help        help for dequeue
  -l, --limit int   Maximum number of messages to dequeue (default 1)
.
.
.
```

### CODE REVIEW

File: cmd/queue.go [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+  ivcap queue delete test_queue`,
+		Args: validateDeleteCommandArgs,
+		RunE: runDeleteQueueCmd,
+	}
+	queueCmd.AddCommand(deleteQueueCmd)
+}
+
+func enqueueCommand() {
+	enqueueCmd := &cobra.Command{
+		Use:   "enqueue [flags] schema queue_id message",
```
Comment (maxott) [inline cmd/queue.go:124]: Can we swap `schema` and `queue_id`? For most commands we usually have the resource id right after the sub command.

I would also ingest the message from a file `-f`. Check commands like `service create` which should have implemented that. One can always get it from the command line via `echo "...message..." | ivcap queue enqueue .. -f -`

File: cmd/queue.go [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+		RunE: runEnqueueCmd,
+	}
+
+	queueCmd.AddCommand(enqueueCmd)
+}
+
+func dequeueCommand() {
+	dequeueCmd := &cobra.Command{
+		Use:   "dequeue [flags] queue_id",
+		Short: "Dequeue messages from a queue",
```
Comment (maxott) [inline cmd/queue.go:141]: Where are the messages written to? Again, you may want to add a `-f` flag to set the destination. `-f -` would be stdout. In the case of multiple messages, the `-f` may need to point to a directory. Not sure if you would want then to write file names to stdout.

Thinking about it, maybe without a `-f` you use a table (which you may have implemented here) with the message content as a column ...

**Review by maxott — CHANGES_REQUESTED** [review]

File: cmd/queue.go (hunk 2 above)
Comment (bclews) [reply cmd/queue.go:141]: > Where are the messages written to? Again, you may want to add a `-f` flag to set the destination. `-f -` would be stdout. In the case of multiple messages, the `-f` may need to point to a directory. Not sure if you would want then to write file names to stdout.
> 
> Thinking about it, maybe without a `-f` you use a table (which you may have implemented here) with the message content as a column ...

I have implemented the standard ivcap-cli console output options. The default option is to print to the console using a table. If json/yaml are chosen then the response is outputted to the console in the desired format.

Adding a `-f` option sounds good for long-running options. Maybe I could add an `-f` option for a single file and a `-d` option for a directory?

File: cmd/queue.go (hunk 1 above)
Comment (maxott) [reply cmd/queue.go:124]: Just thinking, you may want to look at `aspectAddCmd`.  In this case the entityID (aka queueID) is mandatory, the schema is defined either by an optional `--schema` flag or of the message is of type json and has a top-level `$schema` field.

Not sure if you could pretty much reuse most of the code - may require a little refactoring to make it work for both cases - all assuming that the two are really that similar as it looks to me.

Comment (bclews) [reply cmd/queue.go:124]: I appreciate the suggestion, I will take a look and get back to you with any questions I may have.

**Comment by bclews** [comment]
76f5b91e76833acf8e533dc6dbe44ae59403ecd0 updates the `enqueue` and `dequeue` commands to load from and write to files. I need to do more validation on enqueueing from a file, but this was a good point in time to push changes upstream.

Help for **enqueue**:

```
Enqueue a message from a file to the specified queue. The message must be in JSON format.

An example of enqueuing a message to a queue:

  ivcap queue enqueue urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167 urn:ivcap:schema:queue:message.1 message.json

Usage:
  ivcap queue enqueue [flags] queue_id schema file

Arguments:
  queue_id  The unique identifier of the queue from which to dequeue messages.
  schema    The schema of the message to enqueue.
  file      The file containing the message to enqueue. If the message is provided through 'stdin' use '-' as the file name.

Flags:
  -h, --help   help for enqueue
Global Flags:
      --access-token string   Access token to use for authentication with API server [IVCAP_ACCESS_TOKEN]
      --context string        Context (deployment) to use
      --debug                 Set logging level to DEBUG
      --no-history            Do not store history
  -o, --output string         Set format for displaying output [json, yaml]
      --silent                Do not show any progress information
      --timeout int           Max. number of seconds to wait for completion (default 30)
```

Help for **dequeue**:

```
Dequeue messages from the specified queue. The messages will be written to the specified file in JSON format.

An example of dequeuing messages from a queue:

    ivcap queue dequeue urn:ivcap:queue:714e549b-ebab-5dd8-8ebd-2e4b0af76167 messages.json

Usage:
  ivcap queue dequeue [flags] queue_id file

Arguments:
  queue_id  The unique identifier of the queue from which to dequeue messages.
  file      The file to write the dequeued messages to.

Flags:
  -h, --help        help for dequeue
  -l, --limit int   Maximum number of messages to dequeue (default 1)
Global Flags:
      --access-token string   Access token to use for authentication with API server [IVCAP_ACCESS_TOKEN]
      --context string        Context (deployment) to use
      --debug                 Set logging level to DEBUG
      --no-history            Do not store history
  -o, --output string         Set format for displaying output [json, yaml]
      --silent                Do not show any progress information
      --timeout int           Max. number of seconds to wait for completion (default 30)
```

**Review by maxott — APPROVED** [review]

**Pull request merged**